docs/src/inverse_problems/optimization_ode_param_fitting.md

```diff
@@ -0,0 +1,173 @@
+```
+
+!!! note
+    Here, a good exercise is to check the resulting parameter set and note that, while it creates a good fit to the data, it does not actually correspond to the original parameter set. [Identifiability](https://www.sciencedirect.com/science/article/pii/S1364815218307278) is a concept that studies how to deal with this problem.
+
+Say that we instead would like to use the [Broyden–Fletcher–Goldfarb–Shannon](https://en.wikipedia.org/wiki/Broyden%E2%80%93Fletcher%E2%80%93Goldfarb%E2%80%93Shanno_algorithm) algorithm, as implemented by the [NLopt.jl](https://github.com/JuliaOpt/NLopt.jl) package. In this case we would run:
+```@example diffeq_param_estim_1 
+using OptimizationNLopt
+sol = solve(optprob, NLopt.LD_LBFGS())
+nothing # hide
+```
+
+## Optimisation problems with data for multiple species
+Imagine that, in our previous example, we had measurements the concentration of both *S* and *P*:
+```@example diffeq_param_estim_1 
+data_vals_S = (0.8 .+ 0.4*rand(10)) .* data_sol[:S][2:end]
+data_vals_P = (0.8 .+ 0.4*rand(10)) .* data_sol[:P][2:end]
+
+plot(true_sol; idxs=[:S, :P], label=["True S" "True P"], lw=8)
+plot!(data_ts, data_vals_S; label="Measured S", seriestype=:scatter, ms=6, color=:blue)
+plot!(data_ts, data_vals_P; label="Measured P", seriestype=:scatter, ms=6, color=:red)
+```
+
+In this case we would have to use the `L2Loss(data_ts, hcat(data_vals_S, data_vals_P))` and `save_idxs=[1,4]` arguments in `loss_function`:
+```@example diffeq_param_estim_1 
+loss_function_S_P = build_loss_objective(oprob, Tsit5(), L2Loss(data_ts, Array(hcat(data_vals_S, data_vals_P)')), Optimization.AutoForwardDiff(); maxiters=10000, verbose=false, save_idxs=[1,4])
+nothing # hide
+```
+Here, `Array(hcat(data_vals_S, data_vals_P)')` is required to put the data in the right form (in this case, a 2x10 matrix).
+
+We can now fit our model to data and plot the results:
+```@example diffeq_param_estim_1 
+optprob_S_P = OptimizationProblem(loss_function_S_P, [1.0,1.0, 1.0])
+optsol_S_P = solve(optprob_S_P, Optim.NelderMead())
+oprob_fitted_S_P = remake(oprob; p=optsol_S_P.u)
+fitted_sol_S_P = solve(oprob_fitted_S_P, Tsit5())
+plot!(fitted_sol_S_P; idxs=[:S, :P], label="Fitted solution", linestyle=:dash, lw=6, color=[:lightblue :pink])
+```
+
+## Setting parameter constraints and boundaries
+Sometimes, it is desired to set boundaries on parameter values. Indeed, this can speed up the optimisation process (by preventing searching through unfeasible parts of parameter space) or be a prerequisite of some optimisation methods. This can be done by passing the `lb` (lower bounds) and `up` (upper bounds) arguments to `OptimizationProblem`. These are vectors (of the same length as the number of parameters), with each argument corresponding to the boundary value of the parameter with the same index (as used in the `parameters(rn)` vector). If we wish to constrain each parameter to the interval *(0.1, 10.0)* this can be done through:
```

Comment:
```suggestion
Sometimes, it is desired to set boundaries on parameter values. Indeed, this can speed up the optimisation process (by preventing searching through unfeasible parts of parameter space), and can also be a requirement for some optimisation methods. This can be done by passing the `lb` (lower bounds) and `up` (upper bounds) arguments to `OptimizationProblem`. These are vectors (of the same length as the number of parameters), with each argument corresponding to the boundary value of the parameter with the same index (as used in the `parameters(rn)` vector). If we wish to constrain each parameter to the interval $(0.1, 10.0)$ this can be done through:
```